File: app/src/main/java/projects/kyle/ledcontroller/DataHandler/LedDataMock.java
package projects.kyle.ledcontroller.DataHandler;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;
import android.content.Context;
import android.graphics.Color;
import android.net.Uri;
import android.util.Pair;

import java.io.File;
import java.io.IOException;


public class LedDataMock extends ViewModel {

    private MutableLiveData<LedDataHolder> ledData;



    public MutableLiveData<LedDataHolder> getData(Context context){
        if (ledData == null){
            ledData = new MutableLiveData<>();
            intialLoad(context);
        }

        return ledData;
    }

    public boolean saveColorChange(int color, Context context){
        this.getData(context).setValue(new LedDataHolder(color));
        LedDbHelper db = new LedDbHelper(context);
        return db.insertData(color);
    }


    private void intialLoad(Context context){
        LedDataHolder temp = new LedDataHolder();
        temp.color = Color.BLACK;
        ledData.setValue(temp);
        LedDbHelper database = new LedDbHelper(context);

        Pair<Boolean, Integer> dbRead = database.getColor();
        if(dbRead.first){
                ledData.setValue(new LedDataHolder(dbRead.second)); }
        }

 }

File: app/src/main/java/projects/kyle/ledcontroller/DataHandler/LedDataMock.kt
package projects.kyle.ledcontroller.DataHandler


import android.arch.lifecycle.MutableLiveData
import android.arch.lifecycle.ViewModel
import android.content.Context
import android.graphics.Color
import android.util.Pair
import projects.kyle.ledcontroller.Enums.DisplayModeSelection



import projects.kyle.ledcontroller.Models.LedData



class LedDataMock : ViewModel() {

    private var ledData: MutableLiveData<LedData>? = null


    fun getData(context: Context): MutableLiveData<LedData> {
        if (ledData == null) {
            ledData = MutableLiveData()
            intialLoad(context)
        }

        return ledData as MutableLiveData<LedData>
    }


    private fun intialLoad(context: Context) {
        val colorList = arrayListOf<Int>()
        colorList.add(Color.BLACK)

        var temp = LedData(
                colorList = colorList,
                timeMs = 0,
                displayModeSelection = DisplayModeSelection.SOLID)

        ledData?.value = temp

        val database = LedDbHelper(context)

        val dbRead = database.data as Pair<Boolean, LedData>
        if (dbRead.first) {
            ledData?.value = dbRead.second
        }
    }

}
